common/src/Polyhedron_Subtract.h

```diff
@@ -103,610 +67,161 @@ class Polyhedron<T,FP,VP>::Subtract {
-                Polyhedron front = *it;
-                const ClipResult clipResult = front.clip(plane);
-                if (clipResult.success()) {
-                    it->clip(plane.flipped());
-                    it = m_fragments.insert(it, front);
-                    ++it;
-                }
-            }
-            
+            result.push_back(plane);
             currentFace = currentFace->next();
         } while (currentFace != firstFace);
-    }
-    
-    void removeSubtrahend() {
-        const typename V::List vertices = V::asList(m_subtrahend.vertices().begin(), m_subtrahend.vertices().end(), GetVertexPosition());
-        
-        for (auto it = std::begin(m_fragments), end = std::end(m_fragments); it != end; ++it) {
-            const Polyhedron& fragment = *it;
-            if (fragment.hasVertices(vertices, 0.1)) {
-                m_fragments.erase(it);
-                return;
-            }
-        }
-        assert(false);
-    }
-    
-    void simplify() {
-        FragmentVertexSet newFragments = buildNewFragments();
-        removeDuplicateFragments(newFragments);
-        rebuildFragments(newFragments);
-    }
-    
-    FragmentVertexSet buildNewFragments() {
-        FragmentVertexSet result;
-        
-        const ClosestVertices closest = findClosestVertices();
-        if (closest.empty())
-            return findFragmentVertices();
-        
-        for (auto it = std::begin(m_fragments), end = std::end(m_fragments); it != end; ++it) {
-            const Polyhedron& fragment = *it;
-            typename V::Set newFragmentVertices;
-            
-            const Vertex* firstVertex = fragment.vertices().front();
-            const Vertex* currentVertex = firstVertex;
-            
-            do {
-                const V& currentPosition = currentVertex->position();
-                typename ClosestVertices::const_iterator clIt = closest.find(currentPosition);
-                if (clIt != std::end(closest)) {
-                    const V& targetPosition = clIt->second;
-                    newFragmentVertices.insert(targetPosition);
-                } else {
-                    newFragmentVertices.insert(currentPosition);
-                }
-                currentVertex = currentVertex->next();
-            } while (currentVertex != firstVertex);
-            
-            result.insert(newFragmentVertices);
-        }
         
         return result;
     }
     
-    ClosestVertices findClosestVertices() {
-        MoveableVertices moveableVertices = findMoveableVertices();
-        FragmentVertexSet fragmentVertices = findFragmentVertices();
-        return findClosestVertices(moveableVertices, fragmentVertices);
-    }
-    
-    MoveableVertices findMoveableVertices() const {
-        const PositionSet exclude = findExcludedVertices();
-        MoveableVertices result(VertexCmp(0.1));
+    static PlaneList sortPlanes(PlaneList planes) {
```

Comment:
Regarding complexity, it might be quadratic, I haven't analyzed it. It doesn't matter as `n` is usually small. In any case, it's much faster than the old algorithms.